src/qibo/tests/test_qasm.py

```diff
@@ -90,6 +118,20 @@ def test_toffoli():
     assert_strings_equal(c.to_qasm(), target)
 
 
+def test_toffoli_cirq(atol=1e-7):
+    c1 = Circuit(3)
+    c1.add(gates.Y(0))
+    c1.add(gates.TOFFOLI(0, 1, 2))
+    c1.add(gates.X(1))
+    c1.add(gates.TOFFOLI(0, 2, 1))
+    c1.add(gates.Z(2))
+    c1.add(gates.TOFFOLI(1, 2, 0))
+    c2 = circuit_from_qasm(c1.to_qasm())
+    final_state_c1 = c1()
+    final_state_c2 = cirq.Simulator().simulate(c2).final_state

```

Comment:
Just noting that in this circuit only the last Toffoli will act. This is fine because we test both cases of active/inactive controls, I am just mentioning it because sometimes test circuits are trivial and they end up not testing the gate we intend for. A simple way around this is to use a random initial state but this is not required here.